modules/twinklexfd.js

```diff
@@ -689,6 +689,147 @@ Twinkle.xfd.callbacks = {
+		params.logInitialContrib = initialContrib;
+		wikipedia_page.setCallbackParameters(params);
+		wikipedia_page.load(Twinkle.xfd.callbacks.saveLog);
+	},
+	saveLog: function(pageobj) {
+		var text = pageobj.getPageText();
+		var params = pageobj.getCallbackParameters();
+
+		var appendText = '';
+
+		// add blurb if log page doesn't exist
+		if (!pageobj.exists()) {
+			appendText +=
+				"This is a log of all [[WP:XFD|deletion discussion]] nominations made by this user using [[WP:TW|Twinkle]]'s XfD module.\n\n" +
+				'If you no longer wish to keep this log, you can turn it off using the [[Wikipedia:Twinkle/Preferences|preferences panel]], and ' +
+				'nominate this page for speedy deletion under [[WP:CSD#U1|CSD U1]].';
+			if (Morebits.userIsInGroup('sysop')) {
+				appendText += '\n\nThis log does not track XfD-related deletions made using Twinkle.';
+			}
+		}
+
+		// create monthly header
+		var date = new Date(pageobj.getLoadTime());
+		var headerRe = new RegExp('^==+\\s*' + date.getUTCMonthName() + '\\s+' + date.getUTCFullYear() + '\\s*==+', 'm');
+		if (!headerRe.exec(text)) {
+			appendText += '\n\n=== ' + date.getUTCMonthName() + ' ' + date.getUTCFullYear() + ' ===';
+		}
+
+		var editsummary = 'Logging ' + params.venue + ' nomination of [[:' + Morebits.pageNameNorm + ']].';
+		// Provide Wikipedian TLA style: AfD, RfD, CfDS, RM, etc.
+		var toTLACase = function(str) {
+			return str.toString().toUpperCase().replace(/\BF/, 'f');
+		};
+		// If a logged file is deleted but exists on commons, the wikilink will be blue, so provide a link to the log
+		var fileLogLink = mw.config.get('wgNamespaceNumber') === 6 ? ' ([{{fullurl:Special:Log|page=' + mw.util.wikiUrlencode(mw.config.get('wgPageName')) + '}} log])' : '';
+
+		appendText += '\n# [[:' + Morebits.pageNameNorm + ']]' + fileLogLink + ' nominated at [[WP:' + params.venue.toUpperCase() + '|' + toTLACase(params.venue) + ']]';
+		var extraInfo = '';
+
+		switch (params.venue) {
+			case 'afd':
+				if (params.xfdcat) {
+					var afdCats = {
+						'?': 'Unknown',
+						'M': 'Media and music',
+						'O': 'Organisation, corporation, or product',
+						'B': 'Biographical',
+						'S': 'Society topics',
+						'W': 'Web or internet',
+						'G': 'Games or sports',
+						'T': 'Science and technology',
+						'F': 'Fiction and the arts',
+						'P': 'Places and transportation',
+						'I': 'Indiscernible or unclassifiable topic',
+						'U': 'Debate not yet sorted'
+					};
+					extraInfo += '; [[WP:AFDC|Category]]: ' + afdCats[params.xfdcat];
+				}
+				if (params.delsort_cats && params.delsort_cats.length) {
+					extraInfo += '; [[WP:DELSORT|Delsorting]]: ';
+					var delsort_cats = params.delsort_cats;
+					delsort_cats.forEach(function(cat, idx) {
+						delsort_cats[idx] = '[[Wikipedia:WikiProject Deletion sorting/' + cat + '|' + cat + ']]';
+					});
+					extraInfo += delsort_cats.join(', ');
+				}
```

Comment:
Pretty sure all this information isn't worth logging. These would make for a lot of clutter in a XfD log serving little purpose. 

Both of these (esp the latter) are frequently edited by delsort wikignomes after nomination is started (which won't get recorded in the log anyway).

As for `xfdcat`, roughly half of the users don't even bother with filling it (see the counts [here](https://en.wikipedia.org/wiki/Category:AfD_debates)). It is only used for a very broad categorisation of AfDs which I doubt anyone finds useful.

Reply:
Yeah, I don't really disagree.  Mostly put this all in as I'd rather have it all and remove than backfill it later.  It's only potentially interesting, but not hugely useful, especially given the XfD exists so there's a history of it.  Prob excessive here.

Reply:
Removed